cmd/k3s/main.go

```diff
@@ -200,3 +212,33 @@ func extract(dataDir string) (string, error) {
 	}
 	return dir, os.Rename(tempDest, dir)
 }
+
+// findCriConfig returns the path to crictl.yaml
+// crictl won't search multiple locations for a config file. It will fall back to looking in
+// the same directory as the crictl binary, but that's it. We need to check the various possible
+// data-dir locations ourselves and then point it at the right one. We check:
+// - the configured data-dir
+// - the default user data-dir (assuming we can find the user's home directory)
+// - the default system data-dir
+// - the default path from upstream crictl
+func findCriConfig(dataDir string) string {
+	searchList := []string{filepath.Join(dataDir, "agent", criDefaultConfigPath)}
+
+	if homeDataDir, err := resolvehome.Resolve(datadir.DefaultHomeDataDir); err == nil {
+		searchList = append(searchList, filepath.Join(homeDataDir, "agent", criDefaultConfigPath))
+	}
+
+	searchList = append(searchList, filepath.Join(datadir.DefaultDataDir, "agent", criDefaultConfigPath))
+	searchList = append(searchList, criDefaultConfigPath)
+
+	for _, path := range searchList {
+		if _, err := os.Stat(path); err != nil {
+			if errors.Is(err, os.ErrPermission) {
```

Comment:
What's the expected behavior if this isn't a permissions issue?

Reply:
If it's anything else just move on silently - the other most likely error from stat is that the file doesn't exist.

Reply:
Think there'd be any value in logging that if we find a squirrely system?